src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -79,12 +90,27 @@ public ZonedDateTime time() {
         throw new UnsupportedOperationException("time() not yet implemented");
     }
 
-    // @todo #15:30min Implement amount() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
-    public long amount() {
-        throw new UnsupportedOperationException("amount() not yet implemented");
+    public long amount() throws IOException {
+        final String amnt = new IoCheckedScalar<>(
+            () -> new ItemAt<>(
```

Comment:
@carlosmiranda you should be able to directly pass `ItemAt` which implements `Scalar` without using a lambda

Reply:
@victornoel yes, but without the lambda, `amount()` will have to deal with catching the `Exception` thrown by `Scalar.value()`. That, or I make it a  `Scalar<Text>` instead of a `Scalar<String>`, and will have to deal with `Exception` when I call `value().asString()`.

(I love Cactoos, except when it comes to these silly checked exception handling scenarios).

Reply:
@carlosmiranda I would use `UncheckedText` then, because at that point, `ItemAt` has already done its job.

Reply:
@victornoel that doesn't solve the problem, `UncheckedText` cannot accept `ItemAt` directly because it accepts `Text` and not `Scalar`. Which means I have to call `value()`, which means either wrap it in a lambda, or somehow handle `Exception`. Back to square one.

Reply:
@carlosmiranda this works:
```java
        final String amnt = new UncheckedText(
            new IoCheckedScalar<>(
                new ItemAt<>(2, new SplitText(this.transaction, ";"))
            ).value()
        ).asString();
```

Reply:
@victornoel all right, I'll use that.